sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/TextFileFormat.scala

```diff
@@ -99,8 +100,22 @@ class TextFileFormat extends TextBasedFileFormat with DataSourceRegister {
     val broadcastedHadoopConf =
       sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
 
+    val wholeText: Boolean = options.getOrElse("wholetext", "false").toBoolean
```

Comment:
Like what we did for csv and json, could you document this new option in `DataFrameReader`?

Reply:
Move it to `TextOptions`?